src/components/Home.js

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import {connect} from 'react-redux';
+import {withRouter} from 'react-router';
+import {getOffers,getCategories} from '../actions/homeActions';
+import {resetCartData} from '../actions/productActions';
+import SvgLoading from '../presentations/SvgLoading';
+import Header from '../presentations/Header';
+import Footer from '../presentations/Footer';
+import {getViewPortDimensions} from '../utils';
+import LazyLoad, { forceCheck } from 'react-lazyload';
+import Slider from 'react-slick';
+import { Modal } from "react-responsive-modal";
+import Cart from '../presentations/Cart';
+import Constants from '../constants';
+import toastr from 'toastr';
+class Home extends React.Component {
+    constructor(props){
+        super(props);
+        this.state={
+          currentInstance: this,
+          bannerData: [],
+          categoryData: [],
+          dimensions:{},
+          openModal: false,
+          forceCheckSliderImage: false
+        }
+        // this.navigateToPlp = this.navigateToPlp.bind(this);
+        // this.showCartView = this.showCartView.bind(this);
+        // this.onCloseModal = this.onCloseModal.bind(this);
+        // this.checkoutComplete = this.checkoutComplete.bind(this);
+        // this.forceCheckSliderImage = this.forceCheckSliderImage.bind(this);
+    }
+    componentDidMount(){
+      this.props.getOffers();
+      this.props.getCategories();
+    }
+    static getDerivedStateFromProps(props,state){
+      if(props.homeApi.bannerData && props.homeApi.bannerdata_searching_success){
+        state.bannerData = props.homeApi.bannerData;
+        state.currentInstance.forceCheckSliderImage();
```

Comment:
This is where you are making use of class instance saved in the state, so that non-static method could be accessed. This is very wrong.